grpc/interceptors.go

```diff
@@ -83,9 +86,38 @@ func (mc *ModInterceptors) UnaryClientInterceptor(
 	return invoker(ctx, method, req, reply, cc, opts...)
 }
 
-// ModNameUnaryServerInterceptor checks the incoming RPC metadata for a module name and attaches any
+// ModPeerConnTracker is an object (owned by the web service) that manages Module <-> PeerConnection
+// mappings. And provides interceptors for attaching module name and PeerConnection objects onto
+// context's for unary calls.
+type ModPeerConnTracker struct {
+	mu                sync.Mutex
+	modNameToPeerConn map[string]*webrtc.PeerConnection
+}
+
+// NewModPeerConnTracker creates a new ModPeerConnTracker.
+func NewModPeerConnTracker() *ModPeerConnTracker {
+	return &ModPeerConnTracker{
+		modNameToPeerConn: make(map[string]*webrtc.PeerConnection),
+	}
+}
+
+// Add informs the ModPeerConnTracker of a new module name <-> PeerConnection mapping.
+func (tracker *ModPeerConnTracker) Add(modname string, peerConn *webrtc.PeerConnection) {
+	tracker.mu.Lock()
+	tracker.modNameToPeerConn[modname] = peerConn
+	tracker.mu.Unlock()
+}
+
+// Remove removes a mapping from the tracker.
+func (tracker *ModPeerConnTracker) Remove(modname string) {
+	tracker.mu.Lock()
+	delete(tracker.modNameToPeerConn, modname)
+	tracker.mu.Unlock()
+}
+
+// ModInfoUnaryServerInterceptor checks the incoming RPC metadata for a module name and attaches any
 // information to a context that can be retrieved with `GetModuleName`.
-func ModNameUnaryServerInterceptor(
```

Comment:
This interceptor only added on the module name. I figured it'd be less clunky to add the peerconnection attaching into this interceptor, as opposed to creating a new separate interceptor

Reply:
Yeah a second interceptor that would have to:
1. Find the module name (duplicate code to some degree)
2. Lookup the name in the internal mapping of module name string <-> peer connections